### DISCUSSION TITLE
Dialog close by esc-key very slow

### DISCUSSION THREAD

**Discussion by intracloud** [primefaces/community#1402] [PrimeNG]
I'm update from primeng@14.1.2 to primeng@17.10.0

I have a p-table with a icon and on click it show a p-dialog.

Before when press 'esc-key' p-dialgo closed immediately.
Now when I press the ESC key, it takes a long time to close the dialog and it feels like it's hanging.
Investigating it seems that the more records the p-table has, the longer it takes to close the p-dialog.

In version 14.1.2 close immediatley, but in version 15.4.1, 16.9.1 and 17.10.0 close very very slow.

if close by icon-close, top-right p-dialog, ever close quickly.

Thanks.

**Comment by intracloud** [comment]
more info.
if set [closeOnEscape] = "false" not close de p-dialog but still long time blocked.
Thk

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue?

**Comment by mcakici** [comment]
I have the same issue with DialogService.

![image](https://github.com/primefaces/community/assets/3582208/7d69611a-fed2-406f-a58a-9ba5713991d0)

**Comment by mcakici** [comment]
[NgDevTools-Profile-2024-05-08T06_56_24.zip](https://github.com/primefaces/community/files/15244959/NgDevTools-Profile-2024-05-08T06_56_24.zip)
This is the output I got through angular dev tools. When I open a dynamicdialog and press esc, all views are refreshed by angular. How can I ensure that the ESC key remains active only on the relevant active component.

can't create stackblitz demo. can't create the problem when the project is empty.
